### PR TITLE
ARROW-4327: [Python] Add requirements-build.txt convenience file

### PR DESCRIPTION
**Pull request by ryanmwhitephd**
Add additional requirements-build.txt file for setting up a build environment. 
Tested from a centos7 container build (not from an official arrow dockerfile)

### CODE REVIEW

**Review by xhochy — COMMENTED** [review]
Thank you for this PR. This sounds like a good idea. 

@kszucs Is there any place where we could use such a file in the docker setup to see whether this file is still uptodate?

**Comment by ryanmackenziewhite** [comment]
I'll probably need some guidance here to tackle the travis audit failure. Also, if I should try anything else that may be a better solution. Having a way to catch missing dependencies in a docker build would be useful. Right now this file is not actually being tested anywhere b/c the existing reqs files are being used.

**Comment by wesm** [comment]
Can you update the PR title to look like other PRs? "ARROW-$ISSUE_NUMBER: [$COMPONENT] $DESCRIPTION"

File: python/requirements-build.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+six>=1.0.0
```
Comment (xhochy) [inline python/requirements-build.txt]: What about using some of the other requirement files here? You should be able to reference them with
`-r requirements-wheel.txt`

**Review by xhochy — CHANGES_REQUESTED** [review]
To work around the audit failure, you need to add this file to https://github.com/apache/arrow/blob/0c279eb2ce04fa6e8b6051cd25a725dc680a475b/dev/release/rat_exclude_files.txt#L132

**Comment by kszucs** [comment]
Use `pip install -r python/requirements.txt -r python/requirements-test.txt cython` instead.
A `requirements-build.txt` should contain only `cython`.

To maintain successful builds on CentOS we can add a `cpp/Dockerfile.centos` and `python/Dockerfile.centos` based on the alpine dockerfiles then use them from docker-compose like [cpp-alpine](https://github.com/apache/arrow/blob/master/docker-compose.yml#L89) and [python-alpine](https://github.com/apache/arrow/blob/master/docker-compose.yml#L145) respectively.

**Comment by ryanmackenziewhite** [comment]
Seems we have three options here. 
1. Omit `requirements-build.txt`, use `pip install -r python/requirements.txt -r python/requirements-test.txt cython`.
2. Include `cython` only in `requirements-build.txt`, use `pip install -r python/requirements.txt -r python/requirements-test.txt python/requirements-build.txt`
3. Have `requirements-build.txt` use 
```
-r requirements.txt
-r requirements-test.txt
cython
```
Install with `pip install -r requirements-build.txt`

The latter seems the most straightforward for the documentation for getting pyarrow setup (https://arrow.apache.org/docs/python/development.html).

I can look into various dockerfiles from official images if that is useful for arrow to support those for different OSes. I have a mess of different OSes right now that I'm trying to deal with, but all in the RHEL family. I know that the alpine container is a bit different, so that made sense to me to have a dedicated docker.

**Comment by wesm** [comment]
what is the status of this PR?

**Comment by ryanmackenziewhite** [comment]
Just need confirmation what should go in the requirements-build file (only include cython, or depend on the other requirements files), and I'll fix the rat failure as well. I can do that tonight.

**Comment by ryanmackenziewhite** [comment]
@xhochy I updated the requirements-build to depend on the existing req files, and fixed the rat excludes.

File: python/requirements-build.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-r requirements.txt
+-r requirements-test.txt
+cython
+
```
Comment (wesm) [inline python/requirements-build.txt]: I'll remove these superfluous blank lines and merge

**Comment by wesm** [comment]
Argh I am unable to edit this branch anymore because it's "master". I will merge the commit by hand

**Pull request closed without merging**